### PR TITLE
FEAT: custom color for customMessages

### PR DESCRIPTION
**Pull request by Razz21**
validation `customMessages` supports nested objects with `text` / `color` keys

this should work
```js
customMessages: { valueMissing: 'Value is required!'}, // colorInvalid as fallback

customMessages: { valueMissing: { // colorInvalid as fallback
    text: 'Value is required!', 
  } },

customMessages: { valueMissing: { // uses custom color
    text: 'Value is required!',
    color: 'green'
  } },
```

### CODE REVIEW

File: src/components/nCheckboxInput/nCheckboxInput.stories.js [hunk 1]
```diff
@@ -46,7 +46,10 @@ Default.args = {
   id: 'checkbox-input',
   name: 'checkbox-input',
   label: 'Label text',
-  customMessages: { valueMissing: 'Value is required!' },
+  customMessages: { valueMissing: {
+    text: 'Value is required!',
+    color: 'green'
```
Comment (allozaur) [inline src/components/nCheckboxInput/nCheckboxInput.stories.js:51]:
```suggestion
    color: 'pink'
```

**Review by allozaur — CHANGES_REQUESTED** [review]
@Razz21 we still have some imperfections.

1. When there is no value in `pattern` prop we still are getting `patternMismatch` error.

Zrzut ekranu 2021-04-12 o 18 26 12

2. Overriding `typeMismatch` or `patternMismatch` doesn't seem to function properly.

Zrzut ekranu 2021-04-12 o 18 24 22

**Comment by Razz21** [comment]
@allozaur 
1. `nEmailInput` already uses hardcoded `pattern` attribute
obraz
2. second option is to remove `email` type and rely only on regex `pattern` to bypass typeMismatch error
from [MDN](https://developer.mozilla.org/en-US/docs/Web/API/ValidityState/typeMismatch):
```
ValidityState.typeMismatch
The read-only typeMismatch property of a ValidityState object indicates if the value of an <input>, 
after having been edited by the user, does not conform to the constraints set by the element's type attribute.
```

**Comment by allozaur** [comment]
Still couple improvements are needed:

1. Custom `rules` prop doesn't work for:

- Date Input
- Email Input
- Url Input

2. Validation outline colors don't work for

- Checkbox Input
- Date Input
- Radio Input

**Comment by Razz21** [comment]
1. maybe because of this code
```js
const htmlValid = target.checkValidity() // triggers `invalid` input event

// base validation passed
if (htmlValid) {
// check custom validation rules
 this.validateCustomRules(target)
}
```
triggers custom rules **only when** native rules passed

**Review by allozaur — APPROVED** [review]

**Comment by allozaur** [comment]
Alright, so validation logic supporting browser-native validation seems to be working fine. The one exception is Range Input as it potentially utilising more than one `<input type="range">` iniside. This feature will be handled through #66

**Pull request merged**